src/gsy_e/models/strategy/external_strategies/dof_filter.py

```diff
@@ -0,0 +1,24 @@
+"""Module containing filters to be applied to incoming orders received from external strategies."""
+
+from typing import Dict, Tuple, Set
+
+
+class DegreesOfFreedomFilter:
+    """Remove the Degrees of Freedom fields from an incoming order."""
+    FIELDS = ("requirements", "attributes")
+
+    @classmethod
+    def apply(cls, order: Dict) -> Tuple[Dict, Set[str]]:
+        """Remove Degrees of Freedom fields from an order's attributes.
+
+        The order dictionary is modified in-place.
+        Args:
+            order: a dictionary of arguments describing a bid or an offer.
+        """
+        filtered_fields = []
+        for field in cls.FIELDS:
+            if order.get(field):
+                order.pop(field)
+                filtered_fields.append(field)
```

Comment:
How about this?
```suggestion
            if order.pop(field, None):
                filtered_fields.append(field)
```

Reply:
It looks good to me. I remember fiddling with this method for a bit but right now I don't see why we shouldn't use your solution. Thanks!

Reply:
@hannesdiedrich I've followed your comment in 1394ca7